### ISSUE TITLE
attributesToSearchOn fails/panics when searchableAttributes on the index is left at default

### ISSUE THREAD

**Issue by nitrag** [meilisearch/meilisearch#3995]
**Describe the bug**
Searching for specific attributes at search time, using `attributesToSearchOn = ['title']` and the index settings are left at default, will cause a panic. I'm not sure if `showRankingScore: true` is also contributing.

**To Reproduce**
1. Create an index.
2. Add documents.
3. Run a multi-search query:
```python
master_search = {
    'q': keyword,
    'showRankingScore': True,
    'attributesToSearchOn': ['title'],
    'attributesToRetrieve': return_fields,
    'attributesToHighlight': ['title'],
    'highlightPreTag': '<b>',
    'highlightPostTag': '</b>'
}

results = self.client.multi_search(
    [
        {'indexUid': 'table1', **master_search},
        {'indexUid': 'table2', **master_search}
    ]
).get('results')
```
5. Panic!
```
 thread 'tokio-runtime-worker' panicked at 'The ranking rule fid did not sort its bucket exhaustively', milli/src/search/new/bucket_sort.rs:162:13
  stack backtrace:
    0: rust_begin_unwind
             at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/std/src/panicking.rs:579:5
    1: core::panicking::panic_fmt
              at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/core/src/panicking.rs:64:14
    2: milli::search::new::bucket_sort::bucket_sort
    3: milli::search::new::execute_search
    4: milli::search::Search::execute
    5: meilisearch::search::perform_search
    6: tokio::runtime::task::raw::poll
```

**Workaround Steps**
Either of these will work:
- Remove `attributesToSearchOn` from query.
- Create the index, update `update_searchable_attributes` to `['title']`, then Add Documents. Then query no longer needs `attributesToSearchOn` but also keeping it still will not cause the crash it.

**Expected behavior**
A working query.

**Meilisearch version:**
v1.3.1

**Additional context**
This happens locally (Mac M1 Pro, AWS EC2 AMI v1.30). Seems related to https://github.com/meilisearch/meilisearch/issues/3728

**Comment by Kerollmops** [comment]
Hey @nitrag and @GregoryConrad 👋

I want to [release a fix for a similar issue](https://github.com/meilisearch/meilisearch/pull/4013), and I would like to test with your setup: database and query producing the bug. Unfortunately, I don't have your setup. I can only confirm that #4006 is fixed.

I would be glad if you could try this one linked PR code above ☝️, compile it, and try to break it again as you described in both of your issues. Or, if you prefer, give me a dump (yes, it seems sufficient to reproduce) and the query that breaks the engine. This way, I can use my `git bisect`-fu to find what introduced the bug and fix it.

have a nice day 🌞

**Comment by GregoryConrad** [comment]
@Kerollmops I'm guessing you are referring to #3728?

I think my issue there was with my local setup when I thought I deleted an index created with an old version of milli, but didn't actually delete it, and then attempted to open it with a new version of milli. I was never able to reproduce that issue again (even with the same steps I had previously taken), so I am fairly sure it was just a milli version mismatch on my end.

**Comment by Kerollmops** [comment]
Fixed by #4013.

**Comment by nitrag** [comment]
Confirmed fixed! THANK YOU! 🎉
